animation.ts

```diff
@@ -0,0 +1,229 @@
+import "./polyfill/promise";
+import {getStyle, setStyles} from "./dom/css";
+import {merge} from "./extend";
+
+// global map of active element animations
+const elementAnimations = new WeakMap<HTMLElement, AnimationDirector>();
+
+type AnimationOptions = {
```

Comment:
I think we should export all types so other TypeScripts can import and reference them as well. Otherwise it wouldn't be possible, would it?

Reply:
Yes, good idea. At least for not internal-only types.